kasa/smartprotocol.py

```diff
@@ -395,8 +395,12 @@ async def query(self, request: str | dict, retry_count: int = 3) -> dict:
                 ret_val = {}
                 for multi_response in multi_responses:
                     method = multi_response["method"]
-                    self._handle_response_error_code(multi_response, method)
-                    ret_val[method] = multi_response.get("result")
+                    try:
+                        self._handle_response_error_code(multi_response, method)
+                    except KasaException as ex:
+                        ret_val[method] = ex
+                    else:
+                        ret_val[method] = multi_response.get("result")
```

Comment:
This changes the earlier behavior not to crash on errors, which would also fix the issue encountered in #852.